### PR TITLE
docs : add docs for hertz client retry

### PR DESCRIPTION
**Pull request by CSWYF3634076**
#### What type of PR is this?

docs
#### What this PR does / why we need it (en: English/zh: Chinese):

en: add docs for hertz client retry
zh:新增 hertz client retry 文档

### CODE REVIEW

**Comment by li-jin-gou** [comment]
refer to https://github.com/cloudwego/cloudwego.github.io/pull/369 and improve it.

File: content/zh/docs/hertz/tutorials/basic-feature/retry.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+title: "重试"
+date: 2022-10-01
+weight: 10
+description: >
+
+---
+
+hertz为用户提供了自定义的重试逻辑，下面来看一下 client 的 retry 使用方法
```
[the remark this answers is not in the document]
Comment (CSWYF3634076) [reply content/zh/docs/hertz/tutorials/basic-feature/retry.md]: down

File: content/zh/docs/hertz/tutorials/basic-feature/retry.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+## retry次数及延迟配置
```
[the remark this answers is not in the document]
Comment (CSWYF3634076) [reply content/zh/docs/hertz/tutorials/basic-feature/retry.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/retry.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		),
+	)
+}
+```
+
+
+
+## retry条件配置
+
+如果你想要自定义配置重试发生的条件，你可以使用 client.SetRetryIfFunc() 配置，该函数的参数是一个函数 func(req *protocol.Request, resp *protocol.Response, err error) bool 类型，相关参数包括 hertz 请求中的 req、resp 和 err 字段，你可以通过这些参数，判断这个请求该不该重试。在如下例子中，当请求返回的状态码不是 200 或者调用过程中 err != nil 时我们返回 true，即进行重试。
```
[the remark this answers is not in the document]
Comment (CSWYF3634076) [reply content/zh/docs/hertz/tutorials/basic-feature/retry.md]: done

File: content/en/docs/hertz/tutorials/basic-feature/retry.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
+---
+title: "Retry"
+date: 2022-10-01
+weight: 10
+description: >
+
+---
+
+Hertz provides users with customized retry logic，Let's take a look at how to use Client Retry(This section mainly configures the times of retry and the delay policy )
```
Comment (welkeyever) [inline content/en/docs/hertz/tutorials/basic-feature/retry.md:9]: 看着是个中文的逗号呢？好像下面也有几个地方也是，可以在check下哈

Comment (CSWYF3634076) [reply content/en/docs/hertz/tutorials/basic-feature/retry.md]: done

**Review by welkeyever — APPROVED** [review]
LGTM, thx~

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**